code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -81,7 +81,9 @@
 		EscapeConfinement()
 
 	if(AICanContinue(possible_targets))
-		DestroySurroundings()
+		if(!QDELETED(target)) // fix a runtime error, probably hacky
+			if(!targets_from.Adjacent(target)) // so the mob doesnt try to make a path when it can just hit the target
```

Comment:
these two ifs can be combined into one; `if(!QDELETED(target) && !targets_from.Adjacent(target))`

Reply:
can i do the same to these lines
if (iswallturf(T) || ismineralturf(T))
if (!isreinforcedwallturf(T))